8-writing-advanced-tasks/README.md

```diff
@@ -4,251 +4,258 @@
-```
-
-Now let's write the accompanying metadata.  We'll specify the parameters as types such as `"type": "Integer"`  which will help us validate the users input as an `Integer`.  Save the following file to `modules/exercise8/tasks/great_metadata.json`:
-```
-{
-  "description": "An exercise in writing great metadata",
-  "input_method": "stdin",
-
-  "parameters": {
-    "name": {
-      "description": "The description for the 'name' parameter",
-      "type": "String"
-    },
-    "recursive": {
-      "description": "The description for the 'recursive' parameter",
-      "type": "Boolean"
-    },
-    "action": {
-      "description": "The description for the 'action' parameter",
-      "type": "Enum[stop, start, restart]"
-    },
-    "timeout": {
-      "description": "The description for the 'timeout' parameter",
-      "type": "Optional[Integer]"
+# Writing your first task with metadata
+Write a simple task that formats the parameters a user gives it.
+
+1. Save the following file to `modules/exercise8/tasks/great_metadata.py`:
+
+    ```
+    #!/usr/bin/env python
+    
+    """
+    This script prints the values and types passed to it via standard in.  It will
+    return a JSON string with a parameters key containing objects that describe
+    the parameters passed by the user.
+    """
+    
+    import json
+    import sys
+    
+    def make_serializable(object):
+      if isinstance(object, unicode):
+        return object.encode('utf-8')
+      else:
+        return object
+    
+    data = json.load(sys.stdin)
+    
+    message = "Congratulations on writing your metadata!  Here are the keys and the values that you passed to this task."
+    
+    result = {'message': message, 'parameters': []}
+    for key in data:
+        k = make_serializable(key)
+        v = make_serializable(data[key])
+        t = type(data[key]).__name__
+        param = {'key': k, 'value': v, 'type': t}
+        result['parameters'].append(param)
+    
+    print(json.dumps(result))
+    ```
+
+2. Write the accompanying metadata. Specify the parameters as types such as `"type": "Integer"`  which help validate user input as an `Integer`.  
+    ```
+    {
+      "description": "An exercise in writing great metadata",
+      "input_method": "stdin",
+    
+      "parameters": {
+        "name": {
+          "description": "The description for the 'name' parameter",
+          "type": "String"
+        },
+        "recursive": {
+          "description": "The description for the 'recursive' parameter",
+          "type": "Boolean"
+        },
+        "action": {
+          "description": "The description for the 'action' parameter",
+          "type": "Enum[stop, start, restart]"
+        },
+        "timeout": {
+          "description": "The description for the 'timeout' parameter",
+          "type": "Optional[Integer]"
+        }
+      }
     }
-  }
-}
-```
+    ```
+3. Save the file to `modules/exercise8/tasks/great_metadata.json`.
```

Comment:
This is just part of step 2. It's confusing to write the content without knowing where you're writing it to.

Reply:
Fixed.